apps/web/ee/components/workflows/WorkflowDetailsPage.tsx

```diff
@@ -0,0 +1,192 @@
+import { WorkflowActions, WorkflowTemplates } from "@prisma/client";
+import { useRouter } from "next/router";
+import { useState, useEffect, Dispatch, SetStateAction } from "react";
+import { Controller, UseFormReturn } from "react-hook-form";
+
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import { HttpError } from "@calcom/lib/http-error";
+import showToast from "@calcom/lib/notification";
+import { Button } from "@calcom/ui";
+import { Form } from "@calcom/ui/form/fields";
+import { AddActionDialog } from "@ee/components/workflows/AddActionDialog";
+import WorkflowStepContainer from "@ee/components/workflows/WorkflowStepContainer";
+import { Option, FormValues } from "@ee/pages/workflows/[workflow]";
+
+import { trpc } from "@lib/trpc";
+
+import MultiSelectCheckboxes from "@components/ui/form/MultiSelectCheckboxes";
+
+interface Props {
+  form: UseFormReturn<FormValues, any>;
+  workflowId: number;
+  selectedEventTypes: Option[];
+  setSelectedEventTypes: Dispatch<SetStateAction<Option[]>>;
+}
+
+export default function WorkflowDetailsPage(props: Props) {
+  const { form, workflowId, selectedEventTypes, setSelectedEventTypes } = props;
+  const { t } = useLocale();
+  const router = useRouter();
+  const utils = trpc.useContext();
+
+  const [evenTypeOptions, setEventTypeOptions] = useState<Option[]>([]);
+  const [isAddActionDialogOpen, setIsAddActionDialogOpen] = useState(false);
+  const [reload, setReload] = useState(false);
+  const [editCounter, setEditCounter] = useState(0);
+
+  const { data, isLoading } = trpc.useQuery(["viewer.eventTypes"]);
+
+  useEffect(() => {
+    if (data) {
+      let options: Option[] = [];
+      data.eventTypeGroups.forEach((group) => {
+        const eventTypeOptions = group.eventTypes.map((eventType) => {
+          return { value: String(eventType.id), label: eventType.title };
+        });
+        options = [...options, ...eventTypeOptions];
+      });
+      setEventTypeOptions(options);
+    }
+  }, [isLoading]);
```

Comment:
```suggestion
const eventTypeOptions = useMemo(
    () =>
      data?.eventTypeGroups.reduce(
        (options, group) => [
          ...options,
          ...group.eventTypes.map((eventType) => ({
            value: String(eventType.id),
            label: eventType.title,
          })),
        ],
        [] as Option[]
      ) || [],
    [data]
  );
```

We don't need extra state to calculate this, let's just wait for the data and useMemo to prevent re-renders.